testing/fetcher.ts

```diff
@@ -561,12 +563,13 @@ export class AuthenticatingFetcher implements Fetcher {
     }
   }
 
-  private _validateHost(url: string): void {
+  private _validateHost(url: string, method: FetchMethodType): void {
     const parsed = new URL(url);
     const host = parsed.host.toLowerCase();
     const allowedDomains = this._networkDomains || [];
     if (
-      !allowedDomains.map(domain => domain.toLowerCase()).some(domain => host === domain || host.endsWith(`.${domain}`))
+      !allowedDomains.map(domain => domain.toLowerCase()).some(domain => host === domain || host.endsWith(`.${domain}`)) &&
+      !(method === 'GET' ? DEFAULT_ALLOWED_GET_DOMAINS_REGEXES : []).some(domain => host.toLowerCase().match(domain))
```

Comment:
Optional nit, `.test()` returns a boolean so if you're only interested in whether a regex maches, that's preferable.